src/Http/Shared/StreamCopyOperationInternal.cs

```diff
@@ -83,5 +84,57 @@ public static async Task CopyToAsync(Stream source, Stream destination, long? co
                 ArrayPool<byte>.Shared.Return(buffer);
             }
         }
+
+        /// <summary>Asynchronously reads the given number of bytes from the source stream and writes them using pipe writer.</summary>
+        /// <returns>A task that represents the asynchronous copy operation.</returns>
+        /// <param name="source">The stream from which the contents will be copied.</param>
+        /// <param name="writer">The PipeWriter to which the contents of the current stream will be copied.</param>
+        /// <param name="count">The count of bytes to be copied.</param>
+        /// <param name="cancel">The token to monitor for cancellation requests.</param>
+        public static async Task CopyToAsync(Stream source, PipeWriter writer, long? count, CancellationToken cancel)
+        {
+            long? bytesRemaining = count;
+
+            Debug.Assert(source != null);
+            Debug.Assert(writer != null);
+            Debug.Assert(!bytesRemaining.HasValue || bytesRemaining.Value >= 0);
+
+            while (true)
+            {
+                // The natural end of the range.
+                if (bytesRemaining.HasValue && bytesRemaining.Value <= 0)
+                {
+                    return;
+                }
+
+                var memory = writer.GetMemory();
```

Comment:
Better to specify a larger length like in the CopyToAsync stream variant (like up to 85kB) as it will be less iterations of this loop and pipelines now allocates from the ArrayPool and will use it directly without copy; whereas just using `GetMemory()` directly will give back a max of 4kB meaning x20 more syscalls?

Reply:
Something to measure. Now that the pipe can allocate more than 4K it might make sense to do that